### PR TITLE
Fix error formatting to use color defined in `$PSStyle.Formatting`

### LINKED ISSUE #17885: $ErrorView = 'NormalView' leaves the console in Error color after rendering an ErrorRecord
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

Repro steps:
1. Open the latest PowerShell preview.7, and make sure using the default prompt
2. Run `$ErrorView = 'NormalView'`
3. Run `Write-Error 'error'`

This is a bug introduced by https://github.com/PowerShell/PowerShell/pull/17705.
The `Cosnole.ForegroundColor` and `Console.BackgroundColor` were changed but never set back after writing out the message.

https://github.com/PowerShell/PowerShell/blob/1ee27cf7dffeb0175dc42dcd47c39e1725c91b67/src/System.Management.Automation/FormatAndOutput/DefaultFormatters/PowerShellCore_format_ps1xml.cs#L1299-L1307

### Expected behavior

```console
The prompt after executing `Write-Error 'error'` doesn't have the color changed
```

### Actual behavior

```console
The prompt gets changed to the error color
```

### Error details

```console
N/A
```

### Environment data

```powershell
Name                           Value
----                           -----
PSVersion                      7.3.0-preview.7
PSEdition                      Core
GitCommitId                    7.3.0-preview.7
OS                             Microsoft Windows 10.0.19044
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### Visuals

image

### PR DESCRIPTION
**Pull request by daxian-dbw**
# PR Summary

Fix https://github.com/PowerShell/PowerShell/issues/17885, and supersede #17904
Also fix the short-term goal as described in https://github.com/PowerShell/PowerShell/issues/17886#issuecomment-1229015174

1. Make error formatting use colors defined in `$PSStyle.Formatting`, when the host supports virtual terminal.
2. Make `$PSStyle` a constant variable, so it cannot be removed.
3. Make error views other than `ConciseView` to use correct color and reset the color properly when the host supports virtual terminal.

Note that, when the host doesn't support virtual terminal, the error formatting will continue to use color settings in `$Host.PrivateData` to keep consistent with the legacy behavior.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [ ] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.

### CODE REVIEW

File: src/System.Management.Automation/FormatAndOutput/DefaultFormatters/PowerShellCore_format_ps1xml.cs [hunk 1]
```diff
@@ -1278,82 +1254,77 @@ private static IEnumerable<FormatViewDefinition> ViewsOf_System_Management_Autom
...
+                                        return ""${errorColor}$($err.Exception.Message)${resetcolor}""
+                                    }
 
-                                        if ($null -ne $host.PrivateData.ErrorBackgroundColor) {
-                                            [Console]::BackgroundColor = $host.PrivateData.ErrorBackgroundColor
-                                        }
+                                    $myinv = $err.InvocationInfo
+                                    if ($ErrorView -eq 'DetailedView') {
+                                        $message = Get-Error | Out-String
+                                        return ""${errorColor}${message}${resetcolor}""
```
Comment (SteveL-MSFT) [inline src/System.Management.Automation/FormatAndOutput/DefaultFormatters/PowerShellCore_format_ps1xml.cs:1264]: In this case, I think `Get-Error` would have already done it's own error coloring so just `$message` should be sufficient.  Maybe add a comment?

Comment (daxian-dbw) [reply src/System.Management.Automation/FormatAndOutput/DefaultFormatters/PowerShellCore_format_ps1xml.cs:1264]: - When `$PSStyle.OutputRendering` is `Host`, which is the default value, or `PlainText`, `Get-Error | Out-String` will have all ANSI decorations striped off, and hence no color will be applied to the detailed error rendering without adding `$errorColor`.
- When `PSStyle.OutputRendering` is `Ansi`, then the added `$errorColor` will be overridden by the colors from `Get-Error`.

It's a little inconsistent here depending on the value of `OutputRendering`. If we really want it to be consistent, then we can set `$PSStyle.OutputRendering` to either `Host` or `Ansi` (depending on whether we want it to be rendered as error color, or the original color from `Get-Error`), and then revert `OutputRendering` to its original value. Personally, I think this inconsistency is not a big deal; what's your thoughts?

Comment (SteveL-MSFT) [reply src/System.Management.Automation/FormatAndOutput/DefaultFormatters/PowerShellCore_format_ps1xml.cs:1264]: I see.  Perhaps for now what you have is fine.  The extra characters aren't a huge deal.

**Review by SteveL-MSFT — APPROVED** [review]

**Comment by daxian-dbw** [comment]
Mark the PR with `backport-7.2-considered` because https://github.com/PowerShell/PowerShell/issues/17885 is a regression to 7.2 due to the backported PR https://github.com/PowerShell/PowerShell/pull/17705

**Pull request merged**